### PR TITLE
feat: implement separate configuration for resource removal policy

### PR DESCRIPTION
**Pull request by imincik**
This change add separate resources removal policy in order to be able to set it to `RETAIN` for CI deployments which are testing production deployment upgrades.

## Reference

[Code review checklist](CODING.md#Code-review-checklist)

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -174,6 +175,7 @@ jobs:
       run: |
         DEPLOY_ENV="ci${GITHUB_RUN_ID}"
         echo "DEPLOY_ENV=$DEPLOY_ENV" | tee -a $GITHUB_ENV
+        echo "RESOURCE_REMOVAL_POLICY=RETAIN" | tee -a $GITHUB_ENV
```
Comment (l0b0) [inline .github/workflows/deploy.yml:178]: Am I misunderstanding something, or does this make the deployment different from the actual production deployment? Wouldn't that make it likely that `test-prod-copy-upgrade` passes but `deploy` might fail?

**Review by l0b0 — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (imincik) [reply .github/workflows/deploy.yml:178]: `RESOURCE_REMOVAL_POLICY=RETAIN` makes our upgrade testing deployment similar to prod. Until now, removal policy was set to `RETAIN` only for prod deployments.

File: backend/environment.py [hunk 2]
```diff
@@ -1,3 +1,12 @@
 import os
 
+from aws_cdk.core import RemovalPolicy
+
 ENV = os.environ.get("DEPLOY_ENV", "test")
+RESOURCE_REMOVAL_POLICY = os.environ.get("RESOURCE_REMOVAL_POLICY", "DESTROY")
```
Comment (l0b0) [inline backend/environment.py:6]: We can inline this, because the value is only used in one place.

File: backend/environment.py [hunk 3]
```diff
@@ -1,3 +1,12 @@
...
+
+if RESOURCE_REMOVAL_POLICY.upper() == "RETAIN":
+    REMOVAL_POLICY = RemovalPolicy.RETAIN
```
Comment (l0b0) [inline backend/environment.py:9]: We should probably move this to the infrastructure directory, since it's only used there.

**Review by l0b0 — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (l0b0) [reply .github/workflows/deploy.yml:178]: Where is it set to retain prod deployments? `git grep -i RETAIN` doesn't show anything prod-related AFAICT.

File: backend/environment.py (hunk 2 above)
Comment (imincik) [reply backend/environment.py]: Inlined now.

File: backend/environment.py (hunk 3 above)
Comment (imincik) [reply backend/environment.py]: Code moved to `infrastructure/removal_policy.py` file

**Pull request merged**